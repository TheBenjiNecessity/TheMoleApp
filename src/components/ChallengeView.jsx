import React from 'react';

const ChallengeView = ({ challenge }) => {
	return <div />;
};

export default ChallengeView;
