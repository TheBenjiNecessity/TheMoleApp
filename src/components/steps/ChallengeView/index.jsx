import React, { Suspense } from 'react';

const ChallengeView = ({ challenge }) => {
	if (!challenge) return;

	const ChallengeComponent = React.lazy(() => import(`../../challenges/${challenge.type}`));

	return (
		<Suspense fallback={<div>Loading...</div>}>
			<ChallengeComponent challenge={challenge} />
		</Suspense>
	);
};

export default ChallengeView;
