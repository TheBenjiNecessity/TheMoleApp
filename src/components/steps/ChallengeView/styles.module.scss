.top-bar {
	display: flex;
	flex-direction: row;
	padding: 1rem;

	.title {
		flex-grow: 1;
		font-size: 35px;
	}
}
