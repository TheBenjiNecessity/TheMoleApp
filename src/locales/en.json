{
	"common": {
		"or": "Or",
		"ready_players": "Ready Players:"
	},
	"lobby": {
		"roomcode": "Room Code:",
		"player": "Player:",
		"player_plural": "Players:"
	},
	"welcome": {},
	"mole_reveal_view": {
		"title": "Who is the mole?",
		"first_paragraph": "Look at your devices now to see whether or not you have been selected to be the mole.",
		"second_paragraph": "Warning: do not show the other players your role."
	},
	"episode_start": {
		"title": "Episode {{number}}",
		"first_episode_description": "In a moment, each player's name will be input into the input box one at a time. If you see a green screen after your name has been entered you are safe. If, however, you see a red screen after your name has been entered then you have been eliminated from the game."
	},
	"execution_view": {
		"first_paragraph": 
	},
	"execution_wrapup": {
		"title": "Wrap up",
		"executed_player_0": "We're sad to see {{playerName}} go. They will be missed."
	},
	"challenge_intermission": {
		"generic_text": "generic_text"
	},
	"pre_quiz_intermission": {
		"title": "Pre Quiz",
		"first_paragraph": "In a moment, you will take the quiz; ten questions about the identity of the mole. The questions will may be anywhere from \"How tall is the mole?\" to \"What the mole did during a challenge?\".",
		"second_paragraph": "The player who gets the fewest questions correct will be eliminated from the game."
	},
	"quiz": {
		"title": "Quiz",
		"first_paragraph": "Look at your devices now to complete the quiz.",
		"players_done": "Players finished:"
	},
	"post_quiz_intermission": {
		"title": "Post Quiz",
		"first_paragraph": "Everyone has finished their quizes. Let's see who has been eliminated."
	}
}
