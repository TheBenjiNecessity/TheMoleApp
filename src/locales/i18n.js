import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import en from '../locales/en.json';

i18n
	.use(initReactI18next) // passes i18n down to react-i18next
	.init({
		resources: { en },
		ns: 'common',
		defaultNS: 'common',
		fallbackLng: 'en',
		supportedLngs: [ 'en' ],
		interpolation: {
			escapeValue: false
		}
	});

export default i18n;
