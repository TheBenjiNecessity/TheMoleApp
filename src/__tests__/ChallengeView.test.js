import React from 'react';
import { shallow } from 'enzyme';
import '../locales/i18n';

import ChallengeView from '../components/steps/ChallengeView';

describe('ChallengeView', () => {
	it('should render challenge view correctly', () => {
		const challenge = { type: 'outandsafe' };
		const mountedComponent = shallow(<ChallengeView challenge={challenge} />);

		expect(mountedComponent).toMatchSnapshot();
	});
});
