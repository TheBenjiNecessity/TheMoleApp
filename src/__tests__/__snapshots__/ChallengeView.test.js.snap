// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeView should render challenge view correctly 1`] = `
<Suspense
  challenge={
    Object {
      "type": "outandsafe",
    }
  }
  fallback={
    <div>
      Loading...
    </div>
  }
>
  <lazy
    challenge={
      Object {
        "type": "outandsafe",
      }
    }
  />
</Suspense>
`;
